core/types/eth_transaction.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2014 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package types
+
+import (
+	"io"
+	"math/big"
+	"sync/atomic"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/harmony-one/harmony/crypto/hash"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+//go:generate gencodec -type ethTxdata -field-override ethTxdataMarshaling -out gen_eth_tx_json.go
+
+// Shard0ChainID to be reserved unique chain ID for eth compatible chains.
+const Shard0ChainID = 0
```

Comment:
I'm a little confused by this (could be the name?), but is this just a correcting factor to have the chain ID map to a shard ID?

Some clarity on how the other shards would be represented (if at all) would be nice. My guess is that we just bump the eth tx's chain ID by `n` to get to shard `n`?

Reply:
Yes, that's right. will write more comment once we finalized this part of the logic. Now it's just a placeholder.